### PR TITLE
[access-43] Validation: Job Order Module

### PR DESCRIPTION
**Pull request by shernaleeuy**
Done adding validation rules for Job Order Module forms [access-43](https://github.com/dev-tea-party/dev-access/pull/46)

### CODE REVIEW

File: app/Http/Controllers/Projects/JobOrderController.php [hunk 1]
```diff
@@ -46,7 +46,7 @@ public function create()
         $projects = ProjectManage::all();
         $projectcodes = $projects->pluck('prj_code','prj_id');
 
-        return view('job-order.create',compact('projectcodes'));
+    return view('job-order.create',compact('projectcodes'));
```
Comment (rosoarv) [inline app/Http/Controllers/Projects/JobOrderController.php]: don't remove indent

File: app/Http/Controllers/Projects/JobOrderController.php [hunk 2]
```diff
@@ -60,8 +60,26 @@ public function store(Request $request)
     {
         
         $requestData = $request->all();
+        /**
+        */
```
Comment (rosoarv) [inline app/Http/Controllers/Projects/JobOrderController.php]: what is this comment for?

File: app/Http/Controllers/Projects/JobOrderController.php [hunk 3]
```diff
@@ -60,8 +60,26 @@ public function store(Request $request)
...
+            'jo_code.max' => 'The Job Order code may not be greater than 45 characters.',
+            'jo_cost.required'=> 'The Cost field is required.',
+            'jo_cost.max' => 'The Cost may not be greater than 45 characters.',
+            'jo_desc.required' => 'The Description field is required.',
+            'prj_id.required' => "The Project Code field is required.",
+            'prj_id.unique' => "The Project Code attempt to insert duplicate key."
+
+        ]);
         
-        JobOrder::create($requestData);
```
Comment (rosoarv) [inline app/Http/Controllers/Projects/JobOrderController.php]: don't remove the space

File: resources/views/job-order/create.blade.php [hunk 4]
```diff
@@ -27,6 +27,7 @@
 
                         {!! Form::close() !!}
 
+
```
Comment (rosoarv) [inline resources/views/job-order/create.blade.php]: refrain from changing the spaces

**Review by rosoarv — CHANGES_REQUESTED** [review]
only change the codes necessary to the issue

**Comment by shernaleeuy** [comment]
ok sir noted!

**Comment by micosolon** [comment]
cannot make create job order without project code. i cannot proceed

**Comment by jandyl-casas** [comment]
add new project through:
'http://127.0.0.1:8000/projects/project-manage'

File: app/Http/Controllers/Projects/JobOrderController.php [hunk 5]
```diff
@@ -60,8 +60,26 @@ public function store(Request $request)
...
         
         $requestData = $request->all();
+        /**
+        */
+
+        $this->validate($request, [
+            'jo_code' =>'required|max:45',
+            'jo_cost' => 'required|max:45',
+            'jo_desc' => 'required',
+            'prj_id' => 'required|unique:id|numeric'
```
Comment (rosoarv) [inline app/Http/Controllers/Projects/JobOrderController.php]: does prj_id really have to be unique?

**Review by rosoarv — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/Projects/JobOrderController.php (hunk 5 above)
Comment (rosoarv) [reply app/Http/Controllers/Projects/JobOrderController.php]: and also, it'll cause an error after creating a job-order with project id

**Review by micosolon — APPROVED** [review]

**Comment by shernaleeuy** [comment]
Kindly check the updated code for Job Order Controller.  Can now create new job order [access-43](https://github.com/dev-tea-party/dev-access/pull/46
)

**Review by rosoarv — APPROVED** [review]

**Comment by rosoarv** [comment]
@shernaleeuy please merge to master. thanks!

**Pull request merged**